assembly/assemblyCommon/AssemblyAssemblyV2.cc

```diff
@@ -1268,6 +1270,128 @@ void AssemblyAssemblyV2::ApplyPSPToPSSXYOffset_finish()
 }
 // ----------------------------------------------------------------------------------------------------
 
+// ----------------------------------------------------------------------------------------------------
+// Make Space in Z -----------------------------------------------------------------------------------
+// ----------------------------------------------------------------------------------------------------
+void AssemblyAssemblyV2::MakeSpaceOnPlatform_start()
+{
+  if(in_action_){
+
+    NQLog("AssemblyAssemblyV2", NQLog::Warning) << "MakeSpaceOnPlatform_start"
+       << ": logic error, an assembly step is still in progress, will not take further action";
+
+    return;
+  }
+
+  const double dx0 = 0.0;
+  const double dy0 = 0.0;
+  const double dz0 = makespace_Z;
+  const double da0 = 0.0;
+
+  if(dz0 <= 0.)
```

Comment:
Should there be also a check that `makespace_Z` does not exceed the upper stage limit based on the current position? I'm not sure what happens if we try to move up too far and then later (also too far) down. But maybe this is already handled somewhere else

Reply:
Implemented both parts of the solution. I can only test them properly at the setup and will let you know once I did that.

Reply:
It works, I had to solve another issue, though:
`QMessageBox`es should obviously not be called from another thread than the GUI thread. Solved this by emitting a signal to the `AssemblyMainView` and issuing the box there.

Also, we now emit `motion_finished` signals in case the motion could not be performed, which allows us to actually continue using the stage and not being in a fully undefined state :-)
This comes with the following downside: when you try to execute this (or any) step and the stage would need to move beyond its limits, there's a warning popping up, but the step is nevertheless marked as "Done". I will think about a way to resolve this with some kind of `motion_aborted` signal, but that can be a separate MR.